### PR TITLE
Weeks-Chandler-Anderson Potential

### PR DESCRIPTION
**Pull request by fweik**
Fixes me typing `2**(1./6.)` a lot. Is also a little bit faster than `LENNARD_JONES`.

Description of changes:
 - Added an explicit WCA pair potential.

PR Checklist
------------
 - [x] Tests?
   - [x] Interface
   - [x] Core 
 - [x] Docs?

### CODE REVIEW

**Review by hmenke — APPROVED** [review]
Looks good to me.  Just one design question: `LennardJonesInteraction` is spelled out, but `WCAInteraction` is abbreviated.  Should it be more consistent?

**Comment by fweik** [comment]
~~Ah you are probably right. I'll rename it in the interface. In the the stuff is also called lj.~~ 
Nah, `non_bonded_inter[0,0].weeks_chandler_anderson.set_params` is too long. @KaiSzuttor what do you think?

**Comment by KaiSzuttor** [comment]
In the context of interactions i think WCA is a well known abbreviation whereas I would say LJ is not too common. In papers we always talk about WCA interaction but would not write LJ interaction. Thus, I think it's fine as it is.

File: doc/sphinx/inter_non-bonded.rst [hunk 1]
```diff
@@ -216,6 +216,40 @@ interaction, while :math:`\delta` varies how smoothly the potential goes to zero
 alchemical transformations, where a group of atoms can be slowly turned
 on/off during a simulation.
 
+.. _Weeks-Chandler-Anderson interaction:
+
+Weeks-Chandler-Anderson interaction
```
Comment (KaiSzuttor) [inline doc/sphinx/inter_non-bonded.rst]: it's Andersen not Anderson

**Review by KaiSzuttor — CHANGES_REQUESTED** [review]
please change doc for the typo

File: doc/sphinx/inter_non-bonded.rst [hunk 2]
```diff
@@ -216,6 +216,40 @@ interaction, while :math:`\delta` varies how smoothly the potential goes to zero
...
+
+This command defines a Weeks-Chandler-Anderson interaction between particles of the
+types ``type1`` and ``type2``. The potential is defined by
+
+.. math::
+
+   \label{eq:wca}
+     V_\mathrm{WCA}(r) =
+       \begin{cases}
+         4 \epsilon \left[ \left(\frac{\sigma}{r}\right)^{12}
```
Comment (KaiSzuttor) [inline doc/sphinx/inter_non-bonded.rst:242]: It may be handy to allow an ```offset``` similar to the offset in the Lennard-Jones interaction.

Comment (fweik) [reply doc/sphinx/inter_non-bonded.rst:242]: My point was to rather keep it simple, as there is already a Lennard-Jones that has all the bells and whistles.

Comment (KaiSzuttor) [reply doc/sphinx/inter_non-bonded.rst:242]: It's open for extension, so I'm okay

**Review by KaiSzuttor — APPROVED** [review]

**Pull request merged**